File: src/main/java/org/kite9/diagram/visualization/planarization/rhd/position/BoundsBasedPositionRoutingInfo.java
package org.kite9.diagram.visualization.planarization.rhd.position;

import java.awt.geom.Rectangle2D;
import java.util.Map;

import org.kite9.diagram.common.elements.RoutingInfo;
import org.kite9.diagram.common.objects.Bounds;
import org.kite9.diagram.visualization.planarization.mgt.router.RoutableReader.Routing;

public class BoundsBasedPositionRoutingInfo extends PositionRoutingInfo {
	
	Bounds x, y;
	Rectangle2D rect;
	
	public BoundsBasedPositionRoutingInfo(Bounds x, Bounds y) {
		this.x = x;
		this.y = y;
	}

	@Override
	public double centerX() {
		return x.getDistanceCenter();
	}

	@Override
	public double centerY() {
		return y.getDistanceCenter();
	}

	@Override
	public double getMinX() {
		return x.getDistanceMin();
	}

	@Override
	public double getMaxX() {
		return x.getDistanceMax();
	}

	@Override
	public double getMinY() {
		return y.getDistanceMin();
	}

	@Override
	public double getMaxY() {
		return y.getDistanceMax();
	}

	@Override
	public double getWidth() {
		return getMaxX() - getMinX();
	}

	@Override
	public double getHeight() {
		return getMaxY() - getMinY();
	}

	@Override
	public int compareTo(RoutingInfo arg0) {
		BoundsBasedPositionRoutingInfo bbri = (BoundsBasedPositionRoutingInfo) arg0;
		int yc = y.compareTo(bbri.y);
		if (yc != 0) {
			return yc;
		} else {
			int xc = x.compareTo(bbri.x);
			return xc;
		}
	}
	
	private boolean breaksOrder;

	@Override
	public boolean isBreakingOrder() {
		return breaksOrder;
	}

	@Override
	public int compareX(RoutingInfo with) {
		return x.compareTo(((BoundsBasedPositionRoutingInfo)with).x);
	}

	@Override
	public int compareY(RoutingInfo with) {
		return y.compareTo(((BoundsBasedPositionRoutingInfo)with).y);
	}
	
	public Map<Routing, Corner> getAvoidanceCorners() {
		return ac;
	}
	
	private Map<Routing, Corner> ac;
	
	public void setAvoidanceCorners(Map<Routing, Corner> ac) {
		this.ac = ac;
	}
}

File: src/main/kotlin/org/kite9/diagram/visualization/planarization/rhd/position/BoundsBasedPositionRoutingInfo.kt
package org.kite9.diagram.visualization.planarization.rhd.position

import org.kite9.diagram.common.elements.RoutingInfo
import org.kite9.diagram.common.objects.Bounds
import org.kite9.diagram.visualization.planarization.mgt.router.RoutableReader.Routing

class BoundsBasedPositionRoutingInfo(val x: Bounds, val y: Bounds) : PositionRoutingInfo() {

    override fun centerX(): Double {
        return x.distanceCenter
    }

    override fun centerY(): Double {
        return y.distanceCenter
    }

    override fun getMinX(): Double {
        return x.distanceMin
    }

    override fun getMaxX(): Double {
        return x.distanceMax
    }

    override fun getMinY(): Double {
        return y.distanceMin
    }

    override fun getMaxY(): Double {
        return y.distanceMax
    }

    override fun getWidth(): Double {
        return getMaxX() - getMinX()
    }

    override fun getHeight(): Double {
        return getMaxY() - getMinY()
    }


    override fun compareTo(arg0: RoutingInfo): Int {
        val bbri = arg0 as BoundsBasedPositionRoutingInfo
        val yc = y.compareTo(bbri.y)
        return if (yc != 0) {
            yc
        } else {
            x.compareTo(bbri.x)
        }
    }

    override fun compareX(with: RoutingInfo): Int {
        return x.compareTo((with as BoundsBasedPositionRoutingInfo).x)
    }

    override fun compareY(with: RoutingInfo): Int {
        return y.compareTo((with as BoundsBasedPositionRoutingInfo).y)
    }

    var avoidanceCorners: Map<Routing, Corner>? = null

}